rpc/src/services/base_services.rs

```diff
@@ -156,23 +183,30 @@ pub(crate) fn get_block_shell_header(
         .map(|header| BlockHeaderShellInfo::new(&header)))
 }
 
+#[cached(
+    name = "LIVE_BLOCKS_CACHE",
+    type = "TimedSizedCache<(ChainId, BlockHash), Vec<String>>",
```

Comment:
@bkontur about the ones that return `Vec`, is the situation the same for all of them? (empty vec meaning that we don't have data yet, so it shouldn't be cached).

Reply:
@tizoc @mambisi @jurajselep 
well, it depends, there are RPCs like `/mempool/pendings`  which retursn list, and it returns dynamic data, so this should never be cached,

and there are static RPC, which return applied block's data, which are once evaluated, they are unchanged,
but if we have RPC that return Vec and we dont check, if block is applied, but just tries to read some data: `some_storage.get_vec_data(block_hash: &BlockHash) -> Result<Vec<Data>>`,
then we can have problem and cache data, before they are applied, so after apply if we also miss warm-up, we can serve invalid data: empty vs not empty
but in this case, if we `some_storage.get_vec_data` find some Vec data for block, we assume, that have it all and can be cached now